dotnet/src/SemanticKernel.Abstractions/AI/FunctionChoiceBehaviors/FunctionChoiceBehaviorConfiguration.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Represent function choice behavior configuration.
+/// </summary>
+[Experimental("SKEXP0001")]
+public sealed class FunctionChoiceBehaviorConfiguration
+{
+    /// <summary>
+    /// The functions that are available for the model to call.
+    /// </summary>
+    public IEnumerable<KernelFunction>? AvailableFunctions { get; init; }
+
+    /// <summary>
+    /// The functions that the model is required to call.
+    /// </summary>
+    public IEnumerable<KernelFunction>? RequiredFunctions { get; init; }
```

Comment:
Do we ever set both of these?

Can we simplify and just provide the list of functions to be advertised to the LLM?

Reply:
> Do we ever set both of these?

We set either one or another, but never both.

> Can we simplify and just provide the list of functions to be advertised to the LLM?

The list of the functions alone might not be enough. We would also need to pass the choice - auto, required, or none to connectors so they can map it to their model choice.

Reply:
done